lib/modules/manager/gomod/artifacts.ts

```diff
@@ -111,6 +114,107 @@ function useModcacherw(goVersion: string | undefined): boolean {
   );
 }
 
+interface ExtraDep {
+  depName: string;
+  currentValue: string;
+  newValue: string;
+}
+
+function getExtraDeps(
+  goModBefore: string,
+  goModAfter: string,
+  excludeDeps: string[],
+): ExtraDep[] {
+  const result: ExtraDep[] = [];
+
+  const diff = diffLines(goModBefore, goModAfter, {
+    newlineIsToken: true,
+  });
+
+  const addDeps: Record<string, string> = {};
+  const rmDeps: Record<string, string> = {};
+
+  for (const { added, removed, value } of diff) {
+    if (!added && !removed) {
+      continue;
+    }
+
+    const res = parseLine(value);
+    if (!res) {
+      continue;
+    }
+
+    const { depName, currentValue } = res;
+    if (!depName || !currentValue) {
+      continue;
+    }
+
+    if (added) {
+      addDeps[depName] = currentValue;
+    } else {
+      rmDeps[depName] = currentValue;
+    }
+  }
+
+  for (const [depName, currentValue] of Object.entries(rmDeps)) {
+    if (excludeDeps.includes(depName)) {
+      continue;
+    }
```

Comment:
move to loop before to ignore, so this loop becomes cheaper .

what if go changed the version of our updated deps because of other dep constraints? we should also notify that.